bftengine/src/communication/PlainTcpCommunication.cpp

```diff
@@ -561,32 +561,46 @@ class AsyncTcpConnection :
   }
 
  public:
-  void connect(string ip, uint16_t port, bool destIsReplica) {
-    _ip = ip;
+  void connect(string host, uint16_t port, bool destIsReplica) {
+    _host = host;
     _port = port;
     _destIsReplica = destIsReplica;
 
     LOG_TRACE(_logger, "enter, from: " << _selfId
               << " ,to: " << _destId
-              << ", ip: " << ip
+              << ", host: " << host
               << ", port: " << port);
 
-    tcp::endpoint ep(address::from_string(ip), port);
-    LOG_DEBUG(_logger, "connecting from: " << _selfId
-              << " ,to: " << _destId
-              << ", timeout: " << _currentTimeout
-              << ", dest is replica: " << _destIsReplica);
+    asio::ip::tcp::resolver::query query(asio::ip::tcp::v4(), _host, std::to_string(_port));
```

Comment:
It seems pretty much all of these mechanisms are deprecated. I'm not sure how to do things in the recommended manner, but we should probably look into it. https://www.boost.org/doc/libs/1_71_0/doc/html/boost_asio/reference/ip__tcp/resolver.html

Reply:
These are not deprecated in the version of boost that README documents: https://www.boost.org/doc/libs/1_64_0/doc/html/boost_asio/reference/ip__tcp/resolver.html

Nor are they deprecated in the version of boost we actually use: https://www.boost.org/doc/libs/1_65_0/doc/html/boost_asio/reference/ip__tcp/resolver.html

🤷‍♂

Reply:
I guess the deprecation could become a problem if we upgrade Boost versions. Does anyone here have thoughts on whether upgrading Boost versions is something that is likely to happen in the future?

Reply:
Just because we are using an old version doesn't mean the interfaces aren't deprecated. We don't have any immediate plans to upgrade, but if we upgraded we'd likely upgrade to the latest version. If when we do that, the deprecation has taken effect and the methods are removed our code will break. We can push this off until then, when some poor soul will have to fix it, likely due to an unrelated need in a new version of boost.

Reply:
I'm fine merging this in FYI, as long as we all understand this. This is a *very* useful patch.

Reply:
If it is decided to go through with merging this as is and we are concerned about the burden we are creating for some poor soul, I would recommend we immediately create an issue upon merging this to track revising the code to remove the use of deprecated things.

Reply:
I've done more investigation, and we have the standard deprecation problem. The function we're using, `resolve(query&)` is deprecated in a future version, but its replacement `resolve(protocol_type&, string_view, string_view)` is not yet available in the version we're using. So, if we don't want to implement a non-deprecated solution now, we also have to upgrade the boost version now.

The change, when it is needed, should be small: replace the `query` argument to the resolve function with the `ip::tcp::v4`, `_listenHost`, and `_listenPort` arguments that are currently used to construct that query object. What do y'all think about a comment in this code mentioning this, for your "poor soul's" reference?

Reply:
Thanks for investigating Bryan! Let's keep the version as is for now to minimize churn. Let me know when you think this PR is ready to merge.

Reply:
Fixed in bb0343a.